light-prover.js/tests/prover.test.ts

```diff
@@ -0,0 +1,61 @@
+import { it } from "mocha";
+import { Prover } from "../src";
+let circomlibjs = require("circomlibjs");
+import { IDL } from "./circuits/idl";
+
+const chai = require("chai");
+const chaiAsPromised = require("chai-as-promised");
+chai.use(chaiAsPromised);
+
+describe("Prover Functionality Tests", () => {
+  it("Valid proof test", async () => {
+    const poseidon = await circomlibjs.buildPoseidon();
+    const hash = poseidon.F.toString(poseidon([123]));
+    console.log(hash);
+
+    const circuitsPath: string = "./tests/circuits/build-circuits/";
+    const proofInputs: any = {
+      x: 123,
+      hash: hash,
+    };
+
+    const prover = new Prover(IDL, circuitsPath);
+
+    await prover.addProofInputs(proofInputs);
+    await prover.fullProve();
+    await prover.getVkey();
+
+    console.time("Proof generation + Parsing");
+    await prover.fullProveAndParse();
+    console.timeEnd("Proof generation + Parsing");
+    // @ts-ignore
+    globalThis.curve_bn128.terminate();
```

Comment:
what do you think about making this part of the prover class?

Reply:
Agreed, moved to prover.